modules/appeals_api/app/controllers/concerns/appeals_api/caseflow_request.rb

```diff
@@ -0,0 +1,81 @@
+# frozen_string_literal: true
+
+module AppealsApi
+  module CaseflowRequest
+    extend ActiveSupport::Concern
+
+    # Redefine this in the including controller
+    def get_caseflow_response
+      raise 'Unable to get response from caseflow'
+    end
+
+    def caseflow_response
+      @caseflow_response ||= get_caseflow_response
+    rescue Common::Exceptions::BackendServiceException => e
+      log_caseflow_error('BackendServiceException', e)
+      # Try to raise a known type of exception from Common::Exceptions before returning an error response
+      try_to_raise_common_error(e)
+      @caseflow_response = caseflow_exception_to_valid_response(e)
+    end
+
+    # Optionally use this to find SSN for use in the Caseflow request
+    def icn_to_ssn!(icn)
```

Comment:
I don't feel great about putting this method here specifically, but it seemed very useful for all these cases where we're looking up SSN based on ICN, and I don't know of another existing, sharable place to put it.

I think it would be better to put this into a new MPI concern (based on our existing MPIVeteran concern but updated for use with ICNs instead), but I didn't want to add too much to this PR since it's already a lot to digest.